release/build/main.go

```diff
@@ -101,14 +104,15 @@ func hashrelaseSubCommands(cfg *config.Config, runner *registry.DockerRunner) []
 			},
 			Before: func(c *cli.Context) error {
 				configureLogging()
+				// Push the operator hashrelease first, as it is needed for the main hashrelease.
```

Comment:
Why does this need to go in the "before" section? What is this logically changing (other than putting it before the validation)?

Reply:
It needs to happen before validations and since validations are in "Before" that's why I moved it there.
Alternatively, we'll have to move everything in "Before" to "Action"

Reply:
Yeah, I would rather us just keep everything in Action then. I am not sure what splitting into Before() is actually buying us.

I think this is indication that Validate is not factored correctly and might need to be broken down a bit - e.g., "pre-release validation" that validates it's OK to cut a release, and "post-release validation" that validates push happened correctly.

Reply:
Agreed, I will move all to Action.
Once hash release is switched to building its own images, this validation will be obsolete as all that will be needed is image scanning.